qa/tasks/workunit.py

```diff
@@ -16,6 +16,55 @@
 log = logging.getLogger(__name__)
 
 
+class Refspec:
+    def __init__(self, refspec):
+        self.refspec = refspec
+
+    def __str__(self):
+        return self.refspec
+
+    def _clone(self, git_url, clonedir, opts=[]):
```

Comment:
should be opts=None; python binds the default value at definition time, so the list will be be the same for each call, mutated by the call, which is never what you want